docs/INSTALL.md

```diff
@@ -96,6 +95,102 @@ required:
     ```
 
 
+## Configuration: AF_PACKET mode
+
+To configure/install the UPF in AF_PACKET mode, the following changes are required:
+
+1. Enable AF_PACKET mode in configuration file
+
+    ```patch
+    diff --git a/conf/upf.json b/conf/upf.json
+    index 37447b7..b56e1fd 100644
+    --- a/conf/upf.json
+    +++ b/conf/upf.json
+    @@ -1,10 +1,10 @@
+     {
+         "": "Vdev or sim support. Enable `\"mode\": \"af_xdp\"` to enable AF_XDP mode, or `\"mode\": \"af_packet\"` to enable AF_PACKET mode, `\"mode\": \"sim\"` to generate synthetic traffic from BESS's Source module or \"mode\": \"\" when running with UP4",
+         "": "mode: af_xdp",
+    -    "": "mode: af_packet",
+    +    "mode": "af_packet",
+         "": "mode: sim",
+         "": "mode: cndp",
+    -    "mode": "dpdk",
+    +    "": "mode: dpdk",
+
+         "table_sizes": {
+             "": "Example sizes based on sim mode and 50K sessions. Customize as per your control plane",
+    @@ -62,13 +62,13 @@
+
+         "": "Gateway interfaces",
+         "access": {
+    -        "ifname": "ens803f2",
+    +        "ifname": "ens801f0",
+             "": "cndp_jsonc_file: conf/cndp_upf_1worker.jsonc"
+         },
+
+         "": "UE IP Natting. Update the line below to `\"ip_masquerade\": \"<ip> [or <ip>]\"` to enable",
+         "core": {
+    -        "ifname": "ens803f3",
+    +        "ifname": "ens801f1",
+             "": "cndp_jsonc_file: conf/cndp_upf_1worker.jsonc",
+             "": "ip_masquerade: 18.0.0.1 or 18.0.0.2 or 18.0.0.3"
+         },
+    ```
+
+2. Enable AF_PACKET mode and, accordingly, update the script file with the interfaces, IP addresses and MAC addresses including the `ipaddrs` and `nhipaddrs`
+
+    ```patch
+    diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+    index e3ad773..7f1d71a 100755
+    --- a/scripts/docker_setup.sh
+    +++ b/scripts/docker_setup.sh
+    @@ -15,36 +15,36 @@ metrics_port=8080
+     # "af_packet" uses AF_PACKET sockets via DPDK's vdev for pkt I/O.
+     # "sim" uses Source() modules to simulate traffic generation
+     # "cndp" uses kernel AF-XDP. It supports ZC and XDP offload if driver and NIC supports it. It's tested on Intel 800 series n/w adapter.
+    -mode="dpdk"
+    +#mode="dpdk"
+     #mode="cndp"
+     #mode="af_xdp"
+    -#mode="af_packet"
+    +mode="af_packet"
+     #mode="sim"
+
+     # Gateway interface(s)
+     #
+     # In the order of ("s1u" "sgi")
+    -ifaces=("ens803f2" "ens803f3")
+    +ifaces=("ens801f0" "ens801f1")
+
+     # Static IP addresses of gateway interface(s) in cidr format
+     #
+     # In the order of (s1u sgi)
+    -ipaddrs=(198.18.0.1/30 198.19.0.1/30)
```

Comment:
can we maintain the same IP address as mentioned for DPDK mode (like 198.18.0.1/198.19.0.1)? this will give more clarity and restrict only few parameter changes for AF_PACKET mode.